### PR TITLE
{lyn2265} adding a Python API for MotionGroup rules

### PR DESCRIPTION
**Pull request by jackalbe**
{lyn2265} adding a Python API for MotionGroup rules

### CODE REVIEW

File: Gems/PythonAssetBuilder/Editor/Scripts/scene_api/common_rules.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    
+    def to_dict(self):
+        data = vars(self)
+        data['id'] = f"{{{str(self.id)}}}"
+        # rename 'typename' to '$type'
+        data['$type'] = self.typename
+        data.pop('typename')
+        return data
+    
+def convert_rule__to_json(rule:BaseRule, indentValue=0):
```
Comment (AMZN-stankowi) [inline Gems/PythonAssetBuilder/Editor/Scripts/scene_api/common_rules.py:57]: there are two underscores in a row after rule

**Review by AMZN-stankowi — APPROVED** [review]

File: Gems/PythonAssetBuilder/Editor/Scripts/scene_api/common_rules.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
+"""
```
[the remark this answers is not in the document]
Comment (jackalbe) [reply Gems/PythonAssetBuilder/Editor/Scripts/scene_api/common_rules.py:1]: that seems wrong, I will make sure it is correct

File: Gems/PythonAssetBuilder/Editor/Scripts/scene_api/common_rules.py (hunk 1 above)
Comment (jackalbe) [reply Gems/PythonAssetBuilder/Editor/Scripts/scene_api/common_rules.py:57]: yep, will fix

File: Gems/PythonAssetBuilder/Editor/Scripts/scene_api/common_rules.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jackalbe) [reply Gems/PythonAssetBuilder/Editor/Scripts/scene_api/common_rules.py:1]: So I looked around in the <o3de>/scripts folder and it seems there is a mix there. That being said I will attempt to update the headers in the scripts we have in Gems/PythonAssetBuilder folder

**Review by AMZN-stankowi — APPROVED** [review]

**Review by amzn-mike — APPROVED** [review]

**Pull request merged**